elastic-apm-agent/src/main/java/co/elastic/apm/agent/premain/ShadedClassLoader.java

```diff
@@ -219,11 +258,12 @@ public URL findResource(String name) {
      * if the resource could not be found, or if the loader is closed.
      */
     @Override
-    public Enumeration<URL> findResources(String name) throws IOException {
-        if (locallyNonAvailableResources.get().contains(name)) {
-            return null;
+    public Enumeration<URL> findResources(final String name) throws IOException {
+        Enumeration<URL> result = null;
+        if (!locallyNonAvailableResources.get().contains(name)) {
+            result = super.findResources(getShadedResourceName(name));
         }
-        return super.findResources(getShadedResourceName(name));
+        return new PrivilegedEnumeration<URL>(result);
```

Comment:
I think a better alternative is to use the `URLClassLoader` constructor that accepts `AccessControlContext` when constructing this `ShadedClassLoader` with `AccessController.getContext()`. If I understand correctly, it will be automatically used in the `findResource` enumeration.

Reply:
Hmm, that is super odd! 
I wonder if this was just overlooked, or intentional. It seems to completely miss the purpose of the custom `AccessControlContext` and looks very easy to get right. In fact it seems to obvious, that I assume we are missing something here.
Well, if this doesn't help, I guess you need to do what you need to do...

Reply:
The thing is that the internal `URLClassPath` (the `URLClassLoader#ucp` field) is constructed with the provided `AccessControlContext`, so it could have been easily done if `checkURL()` was not a static method.